### PR TITLE
Error 404

### LINKED ISSUE #6: User Story 6, Visitor Navigation Restrictions
As a visitor
When I try to access any path that begins with the following, then I see a 404 error:
- '/merchant'
- '/admin'
- '/profile'

### LINKED ISSUE #7: User Story 7, User Navigation Restrictions
As a default user
When I try to access any path that begins with the following, then I see a 404 error:
- '/merchant'
- '/admin'

### LINKED ISSUE #8: User Story 8, Merchant Navigation Restrictions
As a merchant employee
When I try to access any path that begins with the following, then I see a 404 error:
- '/admin'

### LINKED ISSUE #9: User Story 9, Admin Navigation Restrictions
As an admin
When I try to access any path that begins with the following, then I see a 404 error:
- '/merchant'
- '/cart'

### PR DESCRIPTION
**Pull request by JoseLopez235**
PR Contains:
- All 404 errors for user, merchant, and admin for when they got to a page that they don't belong to
- All Passing Test

closes #6 
closes #7 
closes #8 
closes #9

### CODE REVIEW

File: db/seeds.rb [hunk 1]
```diff
@@ -19,3 +20,10 @@
 #dog_shop items
 pull_toy = dog_shop.items.create(name: "Pull Toy", description: "Great pull toy!", price: 10, image: "http://lovencaretoys.com/image/cache/dog/tug-toy-dog-pull-9010_2-800x800.jpg", inventory: 32)
 dog_bone = dog_shop.items.create(name: "Dog Bone", description: "They'll love it!", price: 21, image: "https://img.chewy.com/is/image/catalog/54226_MAIN._AC_SL1500_V1534449573_.jpg", active?:false, inventory: 21)
+
```
Comment (stillsheryl) [inline db/seeds.rb:23]: Good idea on making seeds! Similarly to my other comment, we may want to make the data different on these to differentiate more easily when running tests.

File: spec/features/navigation/navigation_spec.rb [hunk 2]
```diff
@@ -71,9 +58,126 @@
...
+      expect(page).to have_content("The page you were looking for doesn't exist.")
+
+      visit "/admin"
+      expect(page).to have_content("The page you were looking for doesn't exist.")
+    end
+  end
+
+  describe 'As a merchant' do
+    it "When I try to access any path that begins with the following, then I see a 404 error:" do
+      kiera = User.create!(name: 'Kiera Allen', address: '124 Main St.', city: 'Denver', state: 'CO', zip: 80205, email: 'bob@marley.com', password: 'password', role: 1)
```
Comment (stillsheryl) [inline spec/features/navigation/navigation_spec.rb]: Great job! We may want to stick with different user names for the different roles in our tests so it's easier to tell the difference. For example, in the session new spec we have three different names/info.

for merchant:
`sally = User.create!(name: 'Sally Peach', address: '432 Grove St.', city: 'Denver', state: 'CO', zip: 80205, email: 'sally@peach.com', password: 'password', role: 1)`

for admin:
`bob = User.create!(name: 'Bob Ross', address: '745 Rose St.', city: 'Denver', state: 'CO', zip: 80205, email: 'bob@ross.com', password: 'password', role: 2)`

File: app/assets/stylesheets/application.css [hunk 3]
```diff
@@ -156,3 +156,9 @@ td, th {
 tr:nth-child(even) {
   background-color: #F0F8FF;
 }
+
+.topnav p {
```
Comment (KieraAllen) [inline app/assets/stylesheets/application.css:160]: Thank you, @JoseLopez235 and @c-bartell  for showing me how easy this was!

File: app/controllers/admin/dashboard_controller.rb [hunk 4]
```diff
@@ -4,6 +4,10 @@ class Admin::DashboardController < ApplicationController
   def index
   end
 
+  def users
```
Comment (KieraAllen) [inline app/controllers/admin/dashboard_controller.rb:7]: A `users` method? Hmmm, tell me more!

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -13,9 +13,27 @@
...
+      <% elsif current_user %>
+        <%= link_to "Cart: #{cart.total_items}", "/cart" %>
+        <%= link_to "#{current_user.name}'s Profile", "/profile" %>
+        <%= link_to "Logout", "/logout", method: :delete %>
+        <%= content_tag :p, "Logged in as #{current_user.name}" %>
+      <% else %>
+        <%= link_to "Cart: #{cart.total_items}", "/cart" %>
+        <%= link_to "Register New User", "/register" %>
+        <%= link_to "User Log In", "/login" %>
+      <% end %>
```
Comment (KieraAllen) [inline app/views/layouts/application.html.erb:35]: Looks great! Maybe we can refactor these into partials in the future...

File: config/routes.rb [hunk 6]
```diff
@@ -51,5 +51,6 @@
 
   namespace :admin do
     get "/", to: "dashboard#index"
+    get "/users", to: "dashboard#users"
```
Comment (KieraAllen) [inline config/routes.rb:54]: Ah, so this is where we have the `users` action

File: db/seeds.rb (hunk 1 above)
Comment (KieraAllen) [reply db/seeds.rb:23]: And I could've changed all the names to match the variables (i.e., from 'Kiera Allen' to 'Curtis' and 'Jose')

**Review by KieraAllen — APPROVED** [review]
So it looks like @JoseLopez235 solved the `current_user` problem by creating a `users` action in the Admin Dashboard Controller... I'm excited to hear the process.

File: app/views/layouts/application.html.erb (hunk 5 above)
Comment (c-bartell) [reply app/views/layouts/application.html.erb:35]: That's a great idea!

**Review by c-bartell — APPROVED** [review]

**Pull request merged**